src/hotspot/share/opto/domgraph.cpp

```diff
@@ -410,10 +410,9 @@ void PhaseIdealLoop::Dominators() {
   // Setup mappings from my Graph to Tarjan's stuff and back
   // Note: Tarjan uses 1-based arrays
   NTarjan *ntarjan = NEW_RESOURCE_ARRAY(NTarjan,C->unique()+1);
-  // Initialize _control field for fast reference
-  int i;
-  for( i= C->unique()-1; i>=0; i-- )
-    ntarjan[i]._control = nullptr;
+  // Initialize all fields for safety.
```

Comment:
This is also a performance optimization so the comment should probably say so, so we remember not to change it back to a loop later on.  It might be nice to have NEW_RESOURCE_ARRAY do the initialization for us, like calloc.

Reply:
I mean, this is a common style in C2 to initialize things with `memset`, so I treated it more as "do the same thing as everywhere else", and having perf bump as a nice bonus. I blurped about performance in new comment.